File: apps/server/internal/feed/feed.go
package feed

import "time"

type Post struct {
	ID        string    `json:"id"`
	UserID    string    `json:"user_id"`
	Content   string    `json:"content"`
	CreatedAt time.Time `json:"-"`
	UpdatedAt time.Time `json:"-"`
}

type PostResponse struct {
	ID        string  `json:"id"`
	UserID    string  `json:"user_id"`
	UserName  string  `json:"user_name"`
	Content   string  `json:"content"`
	LikeCount int     `json:"like_count"`
	Liked     bool    `json:"liked"`
	CreatedAt string  `json:"created_at"`
}

type FeedResponse struct {
	Posts      []PostResponse `json:"posts"`
	NextCursor *string        `json:"next_cursor"`
}

type CreatePostRequest struct {
	Content string `json:"content"`
}

type UpdatePostRequest struct {
	Content string `json:"content"`
}

File: apps/mobile/rn/src/infrastructure/api/feed.ts
import {apiClient} from './client';

export interface PostResponse {
  id: string;
  user_id: string;
  user_name: string;
  content: string;
  like_count: number;
  liked: boolean;
  created_at: string;
}

export interface FeedResponse {
  posts: PostResponse[];
  next_cursor: string | null;
}

export function getFeed(cursor?: string, limit = 20) {
  const params = new URLSearchParams();
  if (cursor) {
    params.set('cursor', cursor);
  }
  params.set('limit', String(limit));

  return apiClient.request<FeedResponse>({
    method: 'GET',
    path: `/api/v1/posts?${params.toString()}`,
  });
}

export function getPost(id: string) {
  return apiClient.request<PostResponse>({
    method: 'GET',
    path: `/api/v1/posts/${id}`,
  });
}

export function createPost(content: string) {
  return apiClient.request<PostResponse>({
    method: 'POST',
    path: '/api/v1/posts',
    body: {content},
  });
}

export function updatePost(id: string, content: string) {
  return apiClient.request<PostResponse>({
    method: 'PUT',
    path: `/api/v1/posts/${id}`,
    body: {content},
  });
}

export function deletePost(id: string) {
  return apiClient.request<void>({
    method: 'DELETE',
    path: `/api/v1/posts/${id}`,
  });
}

export function likePost(id: string) {
  return apiClient.request<void>({
    method: 'POST',
    path: `/api/v1/posts/${id}/like`,
  });
}

export function unlikePost(id: string) {
  return apiClient.request<void>({
    method: 'DELETE',
    path: `/api/v1/posts/${id}/like`,
  });
}
